blog/docs/articles/llm-agents-demo.md

```diff
@@ -0,0 +1,136 @@
+spec:
+  reference:
+    apiVersion: serving.knative.dev/v1
+    kind: Service
+    name: average-resource-consumption
+  description: "This tells you the average resource consumption for recent months"
+  type: "average.resource.consumption"
+---
+apiVersion: eventing.knative.dev/v1beta2
+kind: EventType
+metadata:
+  name: resource.cost.calculator
+spec:
+  reference:
+    apiVersion: serving.knative.dev/v1
+    kind: Service
+    name: resource-cost-calculator
+  description: "This calculates the cost for a given resource kind and average usage over a month. Use this tool whenever there is a question about the cost of any cloud resources"
+  schemaData: '{"resourceKind":{"type":"string","description":"The type of resource you would like to calculate the cost of"},"usage":{"type":"float","description":"The average usage of the resource"},"unit":{"type":"string","description":"The unit of the resource usage"}}'
+  type: "resource.cost.calculator"
+```
+
+Both of these eventtypes refer to functions which were deployed in the install
+script we ran earlier. One function returns some mock usage data for the past
+few months, and the other is able to calculate the cost due to the usage of a
+specific resource type for a specific month. After applying both eventtypes,
+we can try asking our original question again:
+
+Calum:
+> How has the trend in the average resource consumption (CPU, memory) changed
+over the past few months for my application, and what impact will this have on
+my costs?
+
+LLM Agent:
+> The trend in the average resource consumption for your application over the
+past few months is as follows:
+
+> - March: CPU - 21.33 Cores, Memory - 8200 MiB
+> - April: CPU - 27.56 Cores, Memory - 8117 MiB
+> - May: CPU - 33.17 Cores, Memory - 9217 MiB
+> - June: CPU - 34.51 Cores, Memory - 10117 MiB
+
+> To determine the impact on your costs, I will calculate the cost for CPU and
+memory based on the average usage over a month for each resource. Let's proceed
+with the cost calculation.The estimated impact on your costs based on the average
+resource consumption for your application over the past few months is as follows:
+
+> - CPU cost: $448.28
+> - Memory cost: $161.87
+
+> These costs are calculated based on the average monthly usage of CPU and memory
+resources. If you have any more questions or need further assistance, feel free
+to ask!
+
+Looking into the tools window, we can see that the LLM called both of the tools
+we provided to it, and the answer is evidently higher quality and more accurate!
+
+What is truly powerful about what we just did was not that the LLM was able to
+use tools to more accurately reply to the user, it was that the LLM was able to
+use tools to more accurately reply to the user _without us re-deploying the LLM
+application_. We have decoupled the development and deployment of the tools from
+the development and deployment of the LLM agent! Now, we are able to declaratively
```

Comment:
Starting from "Now", make it become a separate paragraph would make it easier to read imo.